vortex-file/src/read/layouts/chunked.rs

```diff
@@ -76,38 +79,50 @@ impl ChunkedLayoutBuilder {
     }
 
     fn metadata_layout(&self) -> VortexResult<Option<Box<dyn LayoutReader>>> {
-        self.has_metadata()
-            .then(|| {
+        self.flatbuffer()
+            .metadata()
+            .map(|m| {
+                let set_stats = BooleanBuffer::new(Buffer::from(m.bytes()), 0, Stat::cardinality())
```

Comment:
hm this potentially reads too much if the file has been written on an old version, since we do write all 0s for metadata we could just take the bitlength of metadata bytes here